thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/cache/RedisCacheDAO.java

```diff
@@ -0,0 +1,273 @@
+	 * 
+	 * @param request ThirdEyeCacheRequest. Wrapper for ThirdEyeRequest.
+	 * @return list of TimeSeriesDataPoint
+	 */
+
+	public ThirdEyeCacheResponse tryFetchExistingTimeSeries(ThirdEyeCacheRequest request) throws Exception {
+		long metricId = request.getMetricId();
+		String dimensionKey = request.getDimensionKey();
+		String keyVal = createKey(metricId, dimensionKey);
+		RTimeSeries<Map<String, Object>> ts = getRedissonClient().getTimeSeries(keyVal);
+		ThirdeyeMetricsUtil.redisCallCounter.inc();
+		if (ts.size() == 0) {
+			LOG.info("no redis cache found for window startTime = {} to endTime = {}", request.getStartTimeInclusive(),
+					request.getEndTimeExclusive());
+			ThirdeyeMetricsUtil.redisExceptionCounter.inc();
+			return new ThirdEyeCacheResponse(request, new ArrayList<>());
+		}
+
+		long start = request.getStartTimeInclusive();
+		long end = request.getEndTimeExclusive() - request.getRequest().getGroupByTimeGranularity().toMillis();
+
+		Collection<Map<String, Object>> cacheValues = ts.range(start, end);
+		LOG.info("reading redis cache for window startTime = {} to endTime = {}", request.getStartTimeInclusive(),
+				request.getEndTimeExclusive());
+		List<TimeSeriesDataPoint> timeSeriesRows = new ArrayList<>();
+		for (Map<String, Object> cacheValue : cacheValues) {
+			long timestamp = (long) cacheValue.get(CacheConstants.TIMESTAMP);
+			Double dataValue = (Double) cacheValue.get(CacheConstants.DIMENSION_KEY_VALUE);
+			timeSeriesRows.add(new TimeSeriesDataPoint(request.getMetricUrn(), timestamp, request.getMetricId(),
+					String.valueOf(dataValue)));
+		}
+		return new ThirdEyeCacheResponse(request, timeSeriesRows);
+	}
+
+	/**
+	 * Insert a TimeSeriesDataPoint into Redis. If a document for this data point
+	 * already exists within Redis and the data value for the metricURN already
+	 * exists in the cache, we don't do anything. An example document generated and
+	 * inserted for this might look like: { "timestamp": 123456700000 "metricId":
+	 * 123456, "61427020": "3.0", "83958352": "59.6", "98648743": "0.0" }
+	 * 
+	 * @param point data point
+	 */
+	public void insertTimeSeriesDataPoint(TimeSeriesDataPoint point) {
+		// Check if key (metricId:2:dimensionKey:3158902058) exists
+		// if so, then update - touch/ttl
+		// if not create a key and set data, ttl value
+
+		long timestamp = point.getTimestamp();
+		long metricId = point.getMetricId();
+		String dimensionKey = point.getMetricUrnHash();
+		String keyVal = createKey(metricId, dimensionKey);
+		ThirdeyeMetricsUtil.redisCallCounter.inc();
+
+		if (isActive()) {
+			int ttl = CacheConfig.getInstance().getCentralizedCacheSettings().getTTL();
+			// ts is the key in the redis store. Eg: metricId:2:dimensionKey:3158902058
+			RTimeSeries<Map<String, Object>> ts = getRedissonClient().getTimeSeries(keyVal);
+			Map<String, Object> cacheValue = ts.get(timestamp);
+			if (MapUtils.isEmpty(cacheValue)) {
+				Map<String, Object> values = new HashMap<String, Object>();
```

Comment:
Same here. Where are we using the prefix while setting the metric?

Reply:
`RTimeSeries<Map<String, Object>> ts = getRedissonClient().getTimeSeries(keyVal);`

`keyVal` is the key in the redis